index.html

```diff
@@ -722,47 +1502,116 @@ <h2>
+        unavailable device.
+        </li>
+        <li>Set |gamepad|.{{Gamepad/[[connected]]}} to `false`.
+        </li>
+        <li>Let |navigator:Navigator| be |gamepad|'s [=relevant global
+        object=]'s {{Navigator}} object.
+        </li>
+        <li>If |navigator|.{{Navigator/[[hasGamepadGesture]]}} is `true` and
+        |gamepad|.{{Gamepad/[[exposed]]}} is `true`:
+          <ol>
+            <li>If |gamepad|'s [=relevant global object=] is a {{Window}}
+            object, then [=queue a global task=] on the [=gamepad task source=]
+            to [=fire an event=] named {{gamepaddisconnected}} at |gamepad|'s
+            [=relevant global object=] using {{GamepadEvent}} with its
+            {{GamepadEvent/gamepad}} attribute initialized to |gamepad|.
+            </li>
+          </ol>
+        </li>
+        <li>Set
+        |navigator|.{{Navigator/[[gamepads]]}}[|gamepad|.{{Gamepad/index}}] to
+        `null`.
+        </li>
+        <li>While |navigator|.{{Navigator/[[gamepads]]}} [=list/is not empty=]
+        and the last [=list/item=] of |navigator|.{{Navigator/[[gamepads]]}} is
+        `null`, [=list/remove=] the last [=list/item=] of
+        |navigator|.{{Navigator/[[gamepads]]}}.
+        </li>
+      </ol>
       <p>
         User agents implementing this specification must provide a new DOM
-        event, named <code>gamepaddisconnected</code>. The corresponding event
-        MUST be of type <code>GamepadEvent</code> and, if <a>allowed to use</a>
-        the "`gamepad`" permission, MUST fire on the <code>window</code>
-        object. Registration for and firing of the
-        <code>gamepaddisconnected</code> event MUST follow the usual behavior
-        of DOM Events. [[DOM]]
+        event, named {{gamepaddisconnected}}. The corresponding event MUST be
+        of type {{GamepadEvent}} and, if [=allowed to use=] the `"gamepad"`
+        permission, MUST fire on the {{Window}} object. Registration for and
+        firing of the {{gamepaddisconnected}} event MUST follow the usual
+        behavior of DOM Events. [[DOM]]
       </p>
       <p>
-        When a gamepad is disconnected from the <a>user agent</a>, if the
-        <a>user agent</a> has previously dispatched a <a>gamepadconnected</a>
-        event for that gamepad to a window, a <a>gamepaddisconnected</a> event
-        MUST be dispatched to that same window.
+        When a gamepad is disconnected from the [=user agent=], if the [=user
+        agent=] has previously dispatched a {{gamepadconnected}} event for that
+        gamepad to a window, a {{gamepaddisconnected}} event MUST be dispatched
+        to that same window.
       </p>
       <p>
-        A <a>user agent</a> MUST NOT dispatch this event type if the
-        <a>environment settings object</a> is a <a>non-secure context</a>.
+        A [=user agent=] MUST NOT dispatch this event type if the [=environment
+        settings object=] is a [=non-secure context=].
```

Comment:
Don't need this... can't register events in insecure contexts, so they will never fire in these contexts... 

```suggestion
```

Reply:
Removed